plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/model/StripProductVariationMetaData.kt

```diff
@@ -0,0 +1,32 @@
+package org.wordpress.android.fluxc.model
+
+import com.google.gson.Gson
+import com.google.gson.JsonArray
+import com.google.gson.JsonObject
+import javax.inject.Inject
+
+class StripProductVariationMetaData @Inject internal constructor(private val gson: Gson) {
+    operator fun invoke(metadata: String?): String? {
+        if (metadata == null) return null
+
+        val filteredMetadata = gson.fromJson(metadata, JsonArray::class.java)
+            .mapNotNull { it as? JsonObject }
+            .filter { jsonObject ->
+                val key = jsonObject[WCMetaData.KEY]?.asString ?: ""
+                key in SUPPORTED_KEYS
+            }
+            .filter { jsonObject ->
+                val value = jsonObject[WCMetaData.VALUE]?.asString ?: ""
+                value.isBlank().not()
+            }
+            .toList()
+
+        return if (filteredMetadata.isEmpty()) null else gson.toJson(filteredMetadata)
+    }
```

Comment:
np: since this function has two filter rules, I recommend we verify them at once and operate them with sequences to reduce the number of logical checks we do for each element, maybe something like this:

```suggestion
    gson.fromJson(metadata, JsonArray::class.java)
            .mapNotNull { it as? JsonObject }
            .asSequence()
            .filter { jsonObject ->
                jsonObject[WCMetaData.KEY]?.asString.orEmpty() in SUPPORTED_KEYS &&
                    jsonObject[WCMetaData.VALUE]?.asString.orEmpty().isNotBlank()
            }.toList()
            .takeIf { it.isNotEmpty() }
            ?.let { gson.toJson(it) }
```